executor/analyze_test.go

```diff
@@ -1233,3 +1233,256 @@ func TestAnalyzeSamplingWorkPanic(t *testing.T) {
 	require.NotNil(t, err)
 	require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/executor/mockAnalyzeSamplingMergeWorkerPanic"))
 }
+
+func TestSavedAnalyzeOptions(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("set @@session.tidb_analyze_version = 2")
+	tk.MustExec("set global tidb_persist_analyze_options = true")
```

Comment:
Will the setting operation affect other tests?(I am not sure, it sets a global variable in program) If so, we need to reset it to default value.

Reply:
Fixed.